CMakeLists.txt

```diff
@@ -121,6 +121,11 @@ endif()
 ### Compile- and install-related commands.
 add_subdirectory(src)
 
+# This variable will be filled by the CMakeLists of each library with their
+# target name. The target list will be used to create and install CMake
+# configuration files.
+set(${PROJECT_NAME}_TARGETS)
```

Comment:
I am not a big fan of this "manual" handling of variables and manually dealing with stuff such as `PARENT_SCOPE` , but as far as I understand the alternative is just use a GLOBAL PROPERTY as done in iDynTree, that has the advantages of just requiring a one-iner to add a new variable to the list, see for example https://github.com/robotology/idyntree/blob/2597eac58f87446fd1f6261317a37fd46916a4ad/src/core/CMakeLists.txt#L142, but it no beatiful/superclear either. @drdanz what do you think of this?

Reply:
@traversaro I totally agree with you. As a matter of the fact, I tried to do just like that in the first place 😲, but It wasn't working properly 🙃 So I decided to go with `PARENT_SCOPE`d variable that I agree are super-duper-ugly to have and handle. @drdanz any suggestion? What is the cleanest approach to this?

Reply:
In particular I was not able to pass the targets up in the scope using the `set_property`command and, consequently, [`install_basic_package_files`](https://github.com/robotology/idyntree/blob/2597eac58f87446fd1f6261317a37fd46916a4ad/CMakeLists.txt#L67-L71) was not working properly.

Reply:
I addressed this using a `GLOBAL property`. See new https://github.com/robotology/how-to-export-cpp-library/pull/36/commits/479198992518cc52634fa095dcf717ec6c22267b commit.